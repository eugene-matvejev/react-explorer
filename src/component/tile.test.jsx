import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import _Tile from './tile';

const Tile = (props) =>
    <MemoryRouter>
        <_Tile {...props} />
    </MemoryRouter>;

describe('<Tile/>', () => {
    const props = {
        to: '//example.com',
    };

    describe('render', () => {
        it('with default/required props', () => {
            const { container } = render(<Tile {...props} />);

            expect(container.querySelector('[href="//example.com"]')).toBeInTheDocument();
        });
    });
});
