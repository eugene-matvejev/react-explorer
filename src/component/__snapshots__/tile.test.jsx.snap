// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tile/> render with default/require d props 1`] = `
<Link
  className="tile "
/>
`;

exports[`<Tile/> render with optional props [::className] as "{{className}}" 1`] = `
<Link
  className="tile {{className}}"
/>
`;

exports[`<Tile/> render with optional props [::data-cy] as "{{data-cy}}" 1`] = `
<Link
  className="tile "
  data-cy="{{data-cy}}"
/>
`;
